Jellyfin.Api/Controllers/MediaSegmentsController.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using Jellyfin.Api.Models.MediaSegmentsDtos;
+using Jellyfin.Data.Enums.MediaSegmentType;
+using Jellyfin.Extensions;
+using MediaBrowser.Common.Api;
+using MediaBrowser.Model.MediaSegments;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers;
+
+/// <summary>
+/// Media Segments controller.
+/// </summary>
+[Authorize]
+public class MediaSegmentsController : BaseJellyfinApiController
+{
+    private readonly IMediaSegmentsManager _mediaSegmentManager;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MediaSegmentsController"/> class.
+    /// </summary>
+    /// <param name="mediaSegmentManager">Instance of the <see cref="IMediaSegmentsManager"/> interface.</param>
+    public MediaSegmentsController(
+        IMediaSegmentsManager mediaSegmentManager)
+    {
+        _mediaSegmentManager = mediaSegmentManager;
+    }
+
+    /// <summary>
+    /// Get all media segments.
+    /// </summary>
+    /// <param name="itemId">The item id.</param>
+    /// <param name="streamIndex">Just segments with MediaStreamIndex.</param>
+    /// <param name="type">All segments of type.</param>
+    /// <param name="typeIndex">All segments with typeIndex.</param>
+    /// <response code="200">Segments returned.</response>
+    /// <returns>An <see cref="OkResult"/>containing the found segments.</returns>
+    [HttpGet("{itemId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<ActionResult<IReadOnlyList<MediaSegmentDto>>> GetSegments(
```

Comment:
What happens when itemId is invalid? According to the API spec it should ALWAYS respond with a 200. All of these endpoints have the item id in their route so they should 404 when that itemid is invalid/doesn't exist.